src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1510,51 +1532,42 @@ private String quoteUnixCredentials(String str) {
         return str.replace("'", "'\\''");
     }
 
-    private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"");
-            w.flush();
-        }
-        ssh.setExecutable(true);
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTempFile("pass", ".bat");
+        ssh.write("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"", null);
+        ssh.chmod(0700);
```

Comment:
Moving to workspace@tmp is out of scope for this PR. 
@MarkEWaite tmp file is deleted on finally block, isn't this safe enough ? If you don't think so, can rely on FilePath#createTextTempFile which uses java.io.tmpdir